Code/Framework/AzToolsFramework/Tests/EntityInspectorTests.cpp

```diff
@@ -370,4 +376,209 @@ namespace UnitTest
         // Verify that true is returned here when a system component is editable
         EXPECT_TRUE(AzToolsFramework::ComponentPaletteUtil::ContainsEditableComponents(context, &Filter_IsTestComponent3, AZ::ComponentDescriptor::DependencyArrayType()));
     }
+
+    // helperfunction to reflect serialize for all these components to keep code short
+    template<typename ComponentType>
+    void RegisterSerialize(AZ::ReflectContext* context, bool visible, const char* iconPath, int fixedIndex = -1)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<ComponentType, AZ::Component>();
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                auto reflectionBuilder = editContext->Class<ComponentType>(AZ_STRINGIZE(ComponentType), AZ_STRINGIZE(ComponentType));
+                auto classElement = reflectionBuilder->ClassElement(AZ::Edit::ClassElements::EditorData, "");
+                classElement->Attribute(AZ::Edit::Attributes::AddableByUser, true)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, visible ? AZ::Edit::PropertyVisibility::Show :  AZ::Edit::PropertyVisibility::Hide)
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "Inspector Test Components")
+                    ->Attribute(AZ::Edit::Attributes::Icon, iconPath)
+                    ->Attribute(AZ::Edit::Attributes::ViewportIcon, iconPath);
+
+                if (fixedIndex != -1)
+                {
+                    classElement->Attribute(AZ::Edit::Attributes::FixedComponentListIndex, fixedIndex);
+                }
+            }
+        }
+    }
+    class EditorInspectorTestComponentBase : public AZ::Component
+    {
+    public:
+        // These functions are mandatory to provide but are of no use in this case.
+        void Activate() override {}
+        void Deactivate() override {}
+    };
+    
+    class EditorInspectorTestComponent1 : public EditorInspectorTestComponentBase
+    {
+        public:
+            AZ_COMPONENT(EditorInspectorTestComponent1, "{EF3D8047-4FAA-4615-93E1-C2B5B6EB3C08}", AZ::Component);
+            static void Reflect(AZ::ReflectContext* context)
+            {
+                // A component that is user movable and is visible
+                RegisterSerialize<EditorInspectorTestComponent1>(context, true, "Component1.png");
```

Comment:
Do these png files need to exist or these just placeholder since these are test components?

Reply:
They don't need to exist, and aren't actually ever loaded.